src/Slot-Tests/HistorySlotTest.class.st

```diff
@@ -0,0 +1,79 @@
+Class {
+	#name : #HistorySlotTest,
+	#superclass : #SlotSilentTest,
+	#category : #'Slot-Tests-Examples'
+}
+
+{ #category : #tests }
+HistorySlotTest >> testAddOneElement [
+	| slot instance |
+	slot := #slot1 => HistorySlot.
+	aClass := self make: [ :builder | builder slots: {slot} ].
+
+	self compileAccessorsFor: slot.
+
+	instance := aClass new.
+	instance slot1: 25.
+	self assert: instance slot1 equals: 25
+]
+
+{ #category : #tests }
+HistorySlotTest >> testAddTwoElements [
+	| slot instance |
+	slot := #slot1 => HistorySlot.
+	aClass := self make: [ :builder | builder slots: {slot} ].
+
+	self compileAccessorsFor: slot.
+
+	instance := aClass new.
+	instance slot1: 25.
+	instance slot1: 35.
+	self assert: instance slot1 equals: 35
+]
+
+{ #category : #tests }
+HistorySlotTest >> testAddTwoElementsHistoryRead [
+	| slot instance |
+	slot := #slot1 => HistorySlot.
+	aClass := self make: [ :builder | builder slots: {slot} ].
+
+	self compileAccessorsFor: slot.
+
+	instance := aClass new.
+	instance slot1: 25.
+	instance slot1: 35.
+	self assert: instance slot1 equals: 35.
+	self assert: (slot readHistory: instance) size equals: 3
```

Comment:
#size seems a softer condition than an array of the historic values  
Is it #(25 35 35) or #(35 35 25) ?

Reply:
yes, I made it better... especially as the first value is nil, which is not at all that obvious